preprocessing/nextclade/src/loculus_preprocessing/nextclade.py

```diff
@@ -235,7 +236,105 @@ def run_sort(
+    """
+    Load the nextclade alignment results into the aligned_nucleotide_sequences dict, mapping each
+    accession to a segmentName: NucleotideSequence dictionary.
+    """
+    with open(result_dir_seg + "/nextclade.aligned.fasta", encoding="utf-8") as aligned_nucs:
+        aligned_nuc = SeqIO.parse(aligned_nucs, "fasta")
+        for aligned_sequence in aligned_nuc:
+            aligned_nucleotide_sequences[aligned_sequence.id][segment] = mask_terminal_gaps(
+                str(aligned_sequence.seq)
+            )
+    return aligned_nucleotide_sequences
+
+
+def load_aligned_aa_sequences(
+    result_dir_seg: str,
+    config: Config,
+    aligned_aminoacid_sequences: dict[AccessionVersion, dict[GeneName, AminoAcidSequence | None]],
+) -> dict[AccessionVersion, dict[GeneName, AminoAcidSequence | None]]:
+    """
+    Load the nextclade amino acid alignment results into the aligned_aminoacid_sequences dict, mapping each
+    accession to a geneName: AminoAcidSequence dictionary.
+    """
+    for gene in config.genes:
+        translation_path = result_dir_seg + f"/nextclade.cds_translation.{gene}.fasta"
+        try:
+            with open(translation_path, encoding="utf-8") as aligned_translations:
+                aligned_translation = SeqIO.parse(aligned_translations, "fasta")
+                for aligned_sequence in aligned_translation:
+                    aligned_aminoacid_sequences[aligned_sequence.id][gene] = mask_terminal_gaps(
+                        str(aligned_sequence.seq), mask_char="X"
+                    )
+        except FileNotFoundError:
+            # This can happen if the sequence does not cover this gene
+            logger.debug(
+                f"Gene {gene} not found in Nextclade results expected at: {translation_path}"
+            )
+    return aligned_aminoacid_sequences
+
+
+def add_segment_name_error_warning(
+    entry: UnprocessedEntry,
+    config: Config,
+) -> tuple[
+    Alert,
+    dict[SegmentName, NucleotideSequence | None],
+    dict[SegmentName, NucleotideSequence | None],
+]:
+    unaligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None] = defaultdict()
+    aligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None] = defaultdict()
+    alert: Alert = Alert()
+
+    num_valid_segments = 0
+    num_duplicate_segments = 0
+    for segment in config.nucleotideSequences:
+        unaligned_segment = [
+            data
+            for data in entry.data.unalignedNucleotideSequences
+            if re.match(segment + "$", data, re.IGNORECASE)
+        ]
+        if len(unaligned_segment) > 1:
+            num_duplicate_segments += len(unaligned_segment)
+            alert.errors.append(
+                ProcessingAnnotation.from_single(
+                    segment,
+                    AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                    message="Found multiple sequences with the same segment name.",
+                ),
+            )
```

Comment:
Here one could continue and then not have an elif but I think this code will be removed for multipath anyways

Reply:
exactly lets just ignore for now - this is just to make the large diff slightly less intimidating